### ISSUE TITLE
Orphaned features

### ISSUE THREAD

**Issue by armanforghani** [OrchardCMS/Orchard#6755]
I get some warning message similar to the following in debug log file:
```
2016-04-13 08:07:46,273 [23] Orchard.Environment.ShellBuilders.CompositionStrategy - (null) - WARN Identified 'AdvancedPoll' as an orphaned feature state record in Settings_ShellFeatureRecord.
```

What is the best way to remove all orphaned features?

**Comment by CSurieux** [comment]
I had to go directly in db to remove this...

**Comment by armanforghani** [comment]
So i think we need to create a console command for this.

**Comment by sebastienros** [comment]
We should identify what this means, and how we get in this state to see if we can fix how it happens in the first place.

**Comment by sebastienros** [comment]
Not sure we need a command. If it's a real issue a migration should fix existing records, and we need to fix the bug in the first place. IF it's not a real problem, just live with it, still fix the reason why they are orphaned.

**Comment by armanforghani** [comment]
But living is not possible with several warning in every request. 14 warnings will be added to my log file after any request. Deleting a module may cause this issue or deleting or renaming a feature in manifest file. I agree for fixing the main reason.

**Comment by sebastienros** [comment]
Every request? really ?

**Comment by armanforghani** [comment]
Yes, i get warning for a very large percentage of requests not only during shell startup. But i don't get warning sometimes may be for output cache.

**Comment by Skrypt** [comment]
This happened to me when I deleted a theme or module folder manually "without using the UI to uninstall". Would need to test some things to be able to reproduce. I'm not sure if this is because I had cleaned the database using this #1147 and that it misses some lines to clean also Settings_ShellFeatureRecord wich would lead to having these logs entry about an orphaned feature state record.

**Comment by infofromca** [comment]
i also got this today because  I deleted a theme or module folder manually "without using the UI to uninstall". 
i recreate a dummy module called Orchard.Scheduler, then deleted it on UI, which should be ok.
=========
2016-04-15 18:44:06,085 [49] Orchard.Environment.ShellBuilders.CompositionStrategy - (null) - WARN Identified 'Orchard.Scheduler' as an orphaned feature state record in Settings_ShellFeatureRecord.

**Comment by ViRuSTriNiTy** [comment]
Skrpt described it already, it is caused by manually deleting files without using the UI. I got the same message except for `TheThemeMachine`. I deleted it in the past manually as i did not need it. I recovered the file, clicked uninstall in the Admin area and the log is gone.
